spec/Section 5 -- Validation.md

```diff
@@ -564,6 +564,45 @@ fragment conflictingDifferingResponses on Pet {
 }
 ```
 
+The same is true if a field is designated `Non-Nullable` in an operation. In
+this case, `someValue` could be either a `String` or a `String!` which are two
+different types and therefore can not be merged:
+
+```graphql counter-example
+fragment conflictingDifferingResponses on Pet {
+  ... on Dog {
+    someValue: nickname
+  }
+  ... on Cat {
+    someValue: nickname!
+  }
+}
+```
+
+### Client Controlled Nullability Designator List Dimensions
+
+**Formal Specification**
+
+- For each {field} in the document
+  - Let {fieldDef} be the definition of {field}
+  - Let {fieldType} be the type of {fieldDef}
+  - Let {requiredStatus} be the required status of {field}
+  - Let {designatorDepth} be the number of square bracket pairs in
+    {requiredStatus}
+  - Let {typeDepth} be the number of list dimensions in {fieldType}
```

Comment:
more common term 'rank'

Reply:
number of dimensions is more like matrixes: M[i, j, k] - dimension 3; Graphq does not have these; what we have is rank (I believe)

Reply:
PostgreSQL uses `dimensions` (and does not mention "rank"):

> `array_ndims ( anyarray )` → `integer`
> Returns the number of dimensions of the array.
> `array_ndims(ARRAY[[1,2,3], [4,5,6]])` → `2`

C (and thus probably all C-style languages) uses dimensions; the manual page for arrays doesn't mention the term "rank": https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Multidimensional-Arrays

Haskell seems to use dimensions (this page on arrays doesn't mention "rank"): https://www.haskell.org/tutorial/arrays.html

Fortran uses rank, but defines rank as the "number of dimensions": https://www.ibm.com/docs/en/xffbg/121.141?topic=basics-rank-shape-size-array 

.NET uses rank, but quickly defines it as "number of dimensions": https://docs.microsoft.com/en-us/dotnet/api/system.array.rank?view=net-6.0

I think "number of dimensions" is the most universal term, and "rank" is a shorthand used in some languages.

Reply:
From the .NET link above, note that at least in some languages, jagged arrays, i.e. arrays of arrays that need not be the same length, have rank/dimension of 1 rather than being truly multidimensional. Off the cuff, this is not of major concern, as the rank/dimension referred to seems to be most important there in terms of memory management. Although, I agree that the term "depth" more accurately describes what we have in GraphQL. Perhaps we can define depth even without the use of "dimension" at all.